### ISSUE TITLE
Misspecified var_names

### ISSUE THREAD

**Issue by ColCarroll** [arviz-devs/arviz#326]
In certain cases, if an element of `var_names` is not in the dataset, should we should throw an exception? Right not, it will plot any var_names it can find, and ignore the others. This behavior has actually been kind of nice when plotting a centered eight trace vs a non-centered eight trace, but confusing if you typo one of the var_names.

**Comment by ahartikainen** [comment]
Could it throw warning? Sometimes there are "missing" parameters.

Also can do some word2vec similarity scoring what was probably meant? This could be helpful for complex names?

**Comment by ColCarroll** [comment]
right now the behavior is equivalent to `data.posterior[var_names]` (and specifically, `data.posterior[var_names].sel(**coords)`, though, which I like: i.e., matching xarray's behavior.

**Comment by canyon289** [comment]
Fair point, matching xarrays behavior is good idea! We can focus this discussion on how verbose we want to be to user.

**Comment by michaelraczycki** [comment]
Is it still relevant? I'd like to add this check and warnings if it's still needed

**Comment by OriolAbril** [comment]
Try someting like:

```
idata = az.load_arviz_data("rugby")
az.plot_posterior(idata, var_names=("atts", "unexisting_var"))
```

and see what happens

**Comment by aloctavodia** [comment]
Just to update that the current behavior is to throw a KeyError
